### PR TITLE
nimmi89/c01-aws01

### PR DESCRIPTION
**Pull request by nimmi89**
# Description

Exercise : C01-AWS01

Student Name: Nimmi Varghese

>Creating public and private EC2 instance using AWS CLI.

## Type of change

- [ ] Exercise Submission (one exercise per PR)
  - [ ] This PR is just for **one** exercise of a class
  - [ ] PR name follows the pattern `<github-username>/<exercise-number>`
  - [ ] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-aws01/nimmi89/README.md [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+
+>The command returns:
+{
+    "GroupId": "sg-0619b1745d16f5cb6"
+}
+
+>>>Command to add a rule in that security group that allows SSH access from anywhere.
+aws ec2 authorize-security-group-ingress \
+--group-id sg-0619b1745d16f5cb6 \
+--protocol tcp --port 22 --cidr 0.0.0.0/0
```
Comment (kikobr82) [inline classes/01class/exercises/c01-aws01/nimmi89/README.md:27]: This works, but it's not a good practice to have a rule in your SG with the 0.0.0.0/0 as source(unless this is s public service), specially in a public subnet.
Can you think in a way to make this more secure?

File: classes/01class/exercises/c01-aws01/nimmi89/README.md [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+                        "Code": 16,
+                        "Name": "running"
+
+>>>Command to SSH into the instance with the public ip returned and the key pair you created.
+ssh -i C01-AWS01KeyPair.pem ec2-user@52.64.28.243
+
+Note: If it returns an error ec2-user@52.64.28.243: Permission denied (publickey,gssapi-keyex,gssapi-with-mic)., you need to provide permmissions to read your SSH key
+chmod 400 C01-AWS01KeyPair.pem
+
+Run SSH Command again to get into EC2 Instance successfully!!!!!!!!!!!
```
Comment (kikobr82) [inline classes/01class/exercises/c01-aws01/nimmi89/README.md:259]: YAY!!! Well done!

File: classes/01class/exercises/c01-aws01/nimmi89/README.md [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+```
+
+- Commands to create the second EC2 instance and any additional resource required:
+```
+>>>Command to launch a private instance[no-associate-public-ip-address] into your subnet, using the security group and key pair you've created.The ami id should be according to your region. 
+Can use the same security group,key-pair or create new one .
+aws ec2 run-instances \
+--image-id ami-088ff0e3bde7b3fdf --count 1 \
+--instance-type t2.micro --key-name C01-AWS01KeyPair \
+--security-group-ids sg-0619b1745d16f5cb6
```
Comment (kikobr82) [inline classes/01class/exercises/c01-aws01/nimmi89/README.md:279]: Same as the previous one.. any idea on how you can improve the security in this security group? 
remember you can make a dedicated security group for each instance..

File: classes/01class/exercises/c01-aws01/nimmi89/README.md (hunk 1 above)
Comment (nimmi89) [reply classes/01class/exercises/c01-aws01/nimmi89/README.md]: Yes, especially not at all a good practice in production code. Have authorized the security group to allow SSH from specific IP. Have made the changes

File: classes/01class/exercises/c01-aws01/nimmi89/README.md (hunk 3 above)
Comment (nimmi89) [reply classes/01class/exercises/c01-aws01/nimmi89/README.md]: I have made the changes to allow this security group to allow SSH only from the public instance's security group!

File: classes/01class/exercises/c01-aws01/nimmi89/README.md [hunk 4]
```diff
@@ -21,10 +21,10 @@ aws ec2 create-security-group \
     "GroupId": "sg-0619b1745d16f5cb6"
 }
 
->>>Command to add a rule in that security group that allows SSH access from anywhere.
+>>>Command to add a rule in that security group that allows SSH access from a specific IP address only(enhanced security)
 aws ec2 authorize-security-group-ingress \
 --group-id sg-0619b1745d16f5cb6 \
---protocol tcp --port 22 --cidr 0.0.0.0/0
+--protocol tcp --port 22 --cidr 192.168.64.9/32
```
Comment (caiocezart) [inline classes/01class/exercises/c01-aws01/nimmi89/README.md:27]: are you sure this still works?

Comment (nimmi89) [reply classes/01class/exercises/c01-aws01/nimmi89/README.md]: Have assigned my public IP returned by- curl https://checkip.amazonaws.com!!

**Pull request merged**